packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx

```diff
@@ -136,18 +140,45 @@ const renderRowItem: RenderRowItem<CancelableTask> = (row, index) => {
   );
 };
 
+const parseSelectionData = (
+  value: string | string[] | undefined
+): { type: 'file' | 'folder' | undefined; accept: string | undefined } => {
+  const type =
+    value?.[0] === 'file' || value === 'file'
+      ? 'file'
+      : value?.[0] === 'folder' || value === 'folder'
+      ? 'folder'
+      : undefined;
+
+  const accept = type && Array.isArray(value) ? value[1] : undefined;
+
+  return { type, accept };
+};
+
 export const UploadControls = (): JSX.Element => {
-  const [state, handleUpdateState] = useControl({
+  const [{ history, path }] = useControl({ type: 'NAVIGATE' });
+  const [files, setFiles] = React.useState<File[]>([]);
+  const [fileSelect, handleSelect] = useFileSelect(setFiles);
```

Comment:
I'm not sure if this is capturing the `webkitRelativePath` property that includes the folder names. As a result I think the upload folder action is uploading only individual files and not saving the folders. Should this be happening?

Reply:
That was not in scope of the initial approach for `useFileSelect` but can be added in a seperate PR